### PR TITLE
v2 migration: `guides/debug` to next

### PR DESCRIPTION
**Pull request by vasfvitor**
#### What kind of changes does this PR include?

- New or updated content
- Changes to the docs site code

#### Description

Pages: 
https://tauri.app/v1/guides/debugging/application
https://tauri.app/v1/guides/debugging/clion
https://tauri.app/v1/guides/debugging/vs-code

- Diff - https://github.com/tauri-apps/tauri-docs/commit/f4223b48062b2f3963ccfec0ed284f61d4674205
- Sidebar updated - https://starlight.astro.build/guides/sidebar/#autogenerated-groups
- Headings was changed from "Debugging" to "Debug" in each page title, only because the sidebar entry was already named "Debug" in v2
- Added bash syntax highlight to [this section ](https://tauri.app/v1/guides/debugging/application#rust-console):
v1
image
v2, with syntax highlight:
image

TODO:
- [x] Copy i18n pages (if any)
- [ ] Dark/light mode images, right now they all in light variant.
- [x] Update internal links
- [x] Check if this can be removed or reused `title=".vscode/tasks.json"` in debug/vs-code

### CODE REVIEW

**Comment by lorenzolewis** [comment]
FYI, I just updated out Starlight instance with #1710 that now has support for code block titles: https://starlight.astro.build/guides/authoring-content/#expressive-code-features This might fix a few things you've noted here.

I'll let us work on #1670 first and then can work on this after that is merged if that's okay.

**Comment by vasfvitor** [comment]
yeah I noticed they added expressive code, I was eager to see it being updated here :D, I guess there's a minor issue with our current implementation in commands tabs, I'll open an issue to show what I mean.

it's ok, I'll update what was reviewed over #1670 and then we do the last adjustments after it's closed

File: src/content/docs/guides/debug/clion.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+pnpm vite dev
+```
+
+> Note that CLion currently has no support for background tasks in Run/Debug configurations like VS Code has, so for now you need to manually run the development server.
+
+## Launch a Debugging Session
+
+With the development server running and the Run/Debug Configuration selected in the Switcher, you can now start a new debugging session by clicking Debug. CLion will automatically recognize breakpoints placed in any Rust file in your project.
+
+[core process of your tauri app]: ../../references/architecture/process-model.md#the-core-process
```
Comment (vasfvitor) [inline src/content/docs/guides/debug/clion.mdx]: Currently there's this link that is pending migration of https://tauri.app/v1/references/
Same in `vs-code.mdx`

Comment (vasfvitor) [reply src/content/docs/guides/debug/clion.mdx]: another situation is that some links to docs.rs goes to Tauri 1.x

Comment (lorenzolewis) [reply src/content/docs/guides/debug/clion.mdx]: For the time being let's just point this to the absolute URL so it's not a 404 and it still builds.

**Pull request merged**